tests/nested/core20/gadget-reseal/task.yaml

```diff
@@ -0,0 +1,53 @@
+summary: Check that a gadget refresh reseals
+
+systems: [ubuntu-20.04-64]
+
+execute: |
+    # shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+    # Wait for snapd to be seeded
+    nested_exec sudo snap wait system seed.loaded
+    SEALED_KEY_MTIME_1="$(nested_exec sudo stat --format="%Y" /run/mnt/ubuntu-seed/device/fde/ubuntu-data.sealed-key)"
+    RESEAL_COUNT_1="$(nested_exec sudo cat /var/lib/snapd/device/fde/boot-chains | python3 -m json.tool | grep reseal-count|cut -f2 -d: | tr ',' ' ')"
+    # Install new (unasserted) gadget and wait for reboot/change finishing
+    boot_id="$( nested_get_boot_id )"
+    REMOTE_CHG_ID=$(nested_exec sudo snap install --dangerous /var/lib/snapd/snaps/pc_*.snap --no-wait)
+    # no reboot here, no gadget changes
+    nested_exec sudo snap watch "${REMOTE_CHG_ID}"
+  
+    # nothing in the gadget has changed so no reseal was needed
+    SEALED_KEY_MTIME_2="$(nested_exec sudo stat --format="%Y" /run/mnt/ubuntu-seed/device/fde/ubuntu-data.sealed-key)"
+    test "$SEALED_KEY_MTIME_2" -eq "$SEALED_KEY_MTIME_1"
+    RESEAL_COUNT_2="$(nested_exec sudo cat /var/lib/snapd/device/fde/boot-chains | python3 -m json.tool | grep reseal-count|cut -f2 -d: | tr ',' ' ')"
+    test "$RESEAL_COUNT_2" = "$RESEAL_COUNT_1"
+
+    # create modified boot assets
+    KEY_NAME=$(nested_get_snakeoil_key)
+    SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+    SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+    # ensure clean pc-gadget dir
+    rm -rf pc-gadget
+    snap download --basename=pc --channel="20/edge" pc
+    unsquashfs -d pc-gadget pc.snap
+    echo " " >> pc-gadget/grub-recovery.conf
+    # XXX: this needs a bump everytime we change the gadget, add code to
+    #      do this automatically
+    # XXX2: updateing "ubuntu-boot" here fails with (edition 1->2):
+    # ERROR cannot backup volume structure #0 ("mbr"): cannot find device matching structure #0 ("mbr"): unexpected number of matches (0) for /sys/block/*/ubuntu-data-e4fdfe94-a608-429d-a561-de5dd6a41fdb
```

Comment:
still a bit confused if we hit this error still or not, the test passed but the pr is also not going from edition 1 to edition 2, the pr is going from edition 2 to edition 3?

Reply:
I updated the comment now to hopefully make this clearer

Reply:
MBR has no named partition, so the fallback code path in `gadget` tries to locate the parent block device of the source of /writable mount point. The update code is confused by the fact that /writable is on an encrypted partition, so the underlying device was set up by device mapper, but the mount uses `/dev/by-uuid/ubuntu-data-e4fdfe94-a608-429d-a561-de5dd6a41fdb`.

We should address this at some point. I'll add it to my TODO list.

Reply:
Oh, and a simple sed s/edition: 1/edition: 2/ will bump the edition of the `mbr` structure as it is currently defined in the gadget.yaml for pc.